src/zookeeperLock.ts

```diff
@@ -81,49 +81,48 @@ export class ZookeeperLock extends EventEmitter {
         debuglog(JSON.stringify(this.config));
     }
 
+
     /**
      * create a zookeeper client which powers the lock, done when creating a new lock
      * or zk connection expires
      * @returns {Promise<any>}
      */
-    private createClient() : Promise<any> {
+    private createClient() : Promise<any>|any {
```

Comment:
yeah i just thought you were returning a promise in both cases so was confused. but that makes sense